contracts/RMRK/extension/revealable/RMRKRevealable.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+
+import "./IRMRKRevealable.sol";
+import "./IRMRKRevealer.sol";
+
+pragma solidity ^0.8.21;
+
+/**
+ * @title IRMRKRevealable
+ * @author RMRK team
+ * @notice Interface smart contract of the RMRK Revealable extension. This extension simplifies the process of revealing.
+ */
+abstract contract RMRKRevealable is IRMRKRevealable {
+    address private _revealer;
+
+    /**
+     * @notice Returns the address of the revealer contract
+     */
+    function getRevealer() external view returns (address) {
+        return _revealer;
+    }
+
+    /**
+     * @notice Sets the revealer contract address
+     */
+    function _setRevealer(address revealer) internal {
+        _revealer = revealer;
+    }
+
+    /**
+     * @notice Reveals the assets for the given tokenIds
+     * @param tokenIds The tokenIds to reveal
+     * @dev This method SHOULD be called by the owner or approved for assets
+     * @dev This method SHOULD add the asset to the token and accept it
+     * @dev This method SHOULD get the `assetId` to add and replace from the revealer contract
+     * @dev This `assetId` to replace CAN be 0, meaning that the asset is added to the token without replacing anything
+     * @dev The revealer contract MUST take care of ensuring the `assetId` exists on the contract implementating this interface
+     */
+    function reveal(uint256[] memory tokenIds) external {
+        _checkRevealPermissions(tokenIds);
+        uint256 length = tokenIds.length;
+        (
+            uint64[] memory revealedAssetIds,
+            uint64[] memory assetToReplaceIds
+        ) = IRMRKRevealer(_revealer).getRevealedAssets(tokenIds);
+        for (uint256 i; i < length; ) {
+            _addAndAcceptAssetToToken(
+                tokenIds[i],
+                revealedAssetIds[i],
+                assetToReplaceIds[i]
+            );
```

Comment:
Should this also empty `revealedAssetIds` once it's done adding and accepting them?